OsmAnd/src/net/osmand/plus/configmap/routes/SkiRoutesCard.java

```diff
@@ -0,0 +1,117 @@
+package net.osmand.plus.configmap.routes;
+
+import static net.osmand.osm.OsmRouteType.SKI;
+import static net.osmand.plus.render.RendererRegistry.WINTER_SKI_RENDER;
+
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.LinearLayout.LayoutParams;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.R;
+import net.osmand.plus.activities.MapActivity;
+import net.osmand.plus.configmap.ConfigureMapUtils;
+import net.osmand.plus.dashboard.DashboardOnMap;
+import net.osmand.plus.dialogs.SelectMapStyleBottomSheetDialogFragment;
+import net.osmand.plus.helpers.AndroidUiHelper;
+import net.osmand.plus.routepreparationmenu.cards.MapBaseCard;
+import net.osmand.plus.settings.backend.OsmandSettings;
+import net.osmand.plus.settings.backend.preferences.CommonPreference;
+import net.osmand.plus.utils.AndroidUtils;
+import net.osmand.plus.views.OsmandMapTileView;
+import net.osmand.plus.widgets.dialogbutton.DialogButton;
+import net.osmand.plus.widgets.dialogbutton.DialogButtonType;
+import net.osmand.render.RenderingRulesStorage;
+
+public class SkiRoutesCard extends MapBaseCard {
+
+	private static boolean hideSwitchBanner = false;
+
+	public SkiRoutesCard(@NonNull MapActivity mapActivity) {
+		super(mapActivity);
+	}
+
+	@Override
+	public int getCardLayoutId() {
+		return R.layout.ski_route_card;
+	}
+
+	@Override
+	protected void updateContent() {
+		View mapStyleButton = view.findViewById(R.id.map_style_button);
+
+		TextView mapStyleTitle = mapStyleButton.findViewById(R.id.title);
+		mapStyleTitle.setText(R.string.map_widget_renderer);
+
+		TextView mapStyleDescription = mapStyleButton.findViewById(R.id.description);
+		mapStyleDescription.setText(ConfigureMapUtils.getRenderDescr(app));
+
+		ImageView mapStyleIcon = mapStyleButton.findViewById(R.id.icon);
+		mapStyleIcon.setImageDrawable(app.getUIUtilities().getPaintedIcon(R.drawable.ic_action_map_style, settings.getApplicationMode().getProfileColor(nightMode)));
+
+		mapStyleButton.setOnClickListener(v -> {
+			SelectMapStyleBottomSheetDialogFragment.showInstance(activity.getSupportFragmentManager());
+		});
+
+		mapStyleButton.findViewById(R.id.divider).setVisibility(View.GONE);
+		mapStyleButton.findViewById(R.id.toggle_item).setVisibility(View.GONE);
+
+		View switchBanner = view.findViewById(R.id.switch_banner);
+		if (!hideSwitchBanner) {
+			AndroidUiHelper.updateVisibility(switchBanner, true);
+			setupSwitchBanner();
+		} else {
+			AndroidUiHelper.updateVisibility(switchBanner, false);
+		}
+	}
```

Comment:
banner visibility can be changed inside setupSwitchBanner()

also it is easier to read single line like this - AndroidUiHelper.updateVisibility(switchBanner, !hideSwitchBanner);